Objects/obmalloc.c

```diff
@@ -1206,12 +1206,29 @@ static size_t ntimes_arena_allocated = 0;
 /* High water mark (max value ever seen) for narenas_currently_allocated. */
 static size_t narenas_highwater = 0;
 
-static Py_ssize_t _Py_AllocatedBlocks = 0;
+static Py_ssize_t raw_allocated_blocks;
 
 Py_ssize_t
 _Py_GetAllocatedBlocks(void)
 {
-    return _Py_AllocatedBlocks;
+    Py_ssize_t n = raw_allocated_blocks;
+    /* add up allocated blocks for used pools */
+    for (uint i = 0; i < maxarenas; ++i) {
+        uintptr_t base = arenas[i].address;
```

Comment:
In theory, `base` may not be pool-aligned.  So it needs another line here, like,

    base = (uintptr_t)_Py_ALIGN_UP(base, SYSTEM_PAGE_SIZE);

Reply:
Oops - that should round up to `POOL_SIZE`, not page size.